python/pyarrow/array.pxi

```diff
@@ -1566,6 +1568,9 @@ cdef class Array(_PandasConvertible):
         -------
         array : numpy.ndarray
         """
+        if "numpy" not in sys.modules:
```

Comment:
Why is this not using `np is not None`?
This is problematic because `sys.modules['numpy']` could actually be `None`, forcing NumPy to be considered absent (such as in unit tests).